src/net.cpp

```diff
@@ -229,9 +229,27 @@ std::optional<CAddress> GetLocalAddrForPeer(CNode *pnode)
     return std::nullopt;
 }
 
+/**
+ * If an IPv6 address belongs to the address range used by the CJDNS network and
+ * the CJDNS network is reachable (-cjdnsreachable config is set), then change
+ * the type from NET_IPV6 to NET_CJDNS.
+ * @param[in] service Address to convert.
+ * @return either `addr` or `addr` with modified network.
+ */
+CService MaybeFlipIPv6toCJDNS(const CService& service)
+{
+    CService ret{service};
+    if (ret.m_net == NET_IPV6 && ret.m_addr[0] == 0xfc && IsReachable(NET_CJDNS)) {
+        ret.m_net = NET_CJDNS;
+    }
+    return ret;
+}
+
 // learn a new local address
-bool AddLocal(const CService& addr, int nScore)
+bool AddLocal(const CService& addr_, int nScore)
```

Comment:
In commit ec2dc3915a8cb645afceb0a5a80b539a77e77a4b, 

Not sure I like this addr_ approach.
Don't you think a caller should be aware of the transition to CJDNS?

Reply:
I think the callers of `AddLocal()` do not care that a passed `fc...` address may be interpreted as CJDNS instead of IPv6.

Actually, I started by doing the flip just before `AddLocal()` is called, but ended up with duplicated code all over the place in all callers of `AddLocal()` which prompted me to move that snippet inside `AddLocal()`.

Ideally a CJDNS object should be returned by `LookupIntern()` which is the function which creates the `CNetAddr` objects, so they will not be created as IPv6 and later flipped to CJDNS. However `LookupIntern()` and `WrappedGetAddrInfo()` (which it uses) are defined in `netbase.cpp` and `IsReachable()` (needed to tell whether IPv6 or CJDNS should be created) is in `net.cpp` which would create a circular dependency. Hmm... maybe I should move `IsReachable()` to `netbase.cpp`?

Reply:
>Actually, I started by doing the flip just before AddLocal() is called, but ended up with duplicated code all over the place in all callers of AddLocal() which prompted me to move that snippet inside AddLocal().

I think this is a good move, i was just wondering whether it's ok to do it quietly.
Imagine logging something like "finished processing IPv6 addr" after calling AddLocal, while it's not actually IPv6. 

I think since this is not really the case now, making a comment along AddrLocal might be sufficient to prevent the confusion.

>Hmm... maybe I should move IsReachable() to netbase.cpp?

Perhaps attempt that in a separate PR then?